File: 118-pascals-triangle/118-pascals-triangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
    
     let result=[];
    for(i=0;i<numRows;i++){
        let item=[];
let preResultElement = result[i-1];
let len = preResultElement && preResultElement.length || 0;
item.push(1);
if(len > 0){
       for(j=1;j<len;j++){
    item.push(preResultElement[j] + preResultElement[j-1])
}
item.push(1);
}
        result.push(item);
    }
    return result;
};

File: 118-pascals-triangle/118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) 
    {
      List<List<Integer>> result = new ArrayList<>();

        for(int i = 0;i<numRows;i++)
        {
            List<Integer> row = new ArrayList<>();
            List<Integer> prevRow = new ArrayList<>();
            int len = 0;
            if(result.size() > 0)
            {
                len = result.size();
                prevRow = result.get(i-1);
            }
                
            row.add(1);
            
            if(len > 0)
            {
                for(int j=1;j<len;j++)
                {
                    row.add(prevRow.get(j-1)+prevRow.get(j));
                }
                row.add(1);
            }

            result.add(row);
        }

        return result;
    }
}